front/lib/resources/membership_resource.ts

```diff
@@ -424,6 +415,7 @@ export class MembershipResource extends BaseResource<MembershipModel> {
         { role: newRole },
         { where: { id: membership.id }, transaction }
       );
+      Object.assign(membership, { role: newRole });
```

Comment:
Will do a PR once this one is merged to have a method on the base resource.

Reply:
This one is not really required. I can totaly remove it as I chnaged the return type to return the things that changed for analytics reporting

Reply:
(removed)